.github/workflows/template.yml

```
@@ -76,26 +74,46 @@ jobs:
           path: lesson
           fetch-depth: 0
 
-      - name: Determine the proper reference to use
-        id: styles-ref
+      - name: Sync lesson with carpentries/styles
+        working-directory: lesson
         run: |
-          if [[ -n "${{ github.event.pull_request.number }}" ]]; then
-            echo "::set-output name=ref::refs/pull/${{ github.event.pull_request.number }}/head"
+          if [[ -n "${{ github.event.pull_request.number }}" ]]
+          then
+            ref="refs/pull/${{ github.event.pull_request.number }}/head"
           else
-            echo "::set-output name=ref::gh-pages"
+            ref="gh-pages"
           fi
 
-      - name: Sync lesson with carpentries/styles
-        if: ${{ ! matrix.remote-theme }}
-        working-directory: lesson
-        run: |
           git config --global user.email "team@carpentries.org"
           git config --global user.name "The Carpentries Bot"
+
           git remote add styles https://github.com/carpentries/styles.git
-          git config --local remote.styles.tagOpt --no-tags
-          git fetch styles ${{ steps.styles-ref.outputs.ref }}:styles-ref
-          git merge -s recursive -Xtheirs --no-commit styles-ref
-          git commit -m "Sync lesson with carpentries/styles"
+          git fetch styles $ref:styles-ref
+
+          # Sync up only if necessary
+          if [[ $(git rev-list --count HEAD..styles-ref) != 0 ]]
+          then
+
+            # The merge command below might fail for lessons that use remote theme
+            # https://github.com/carpentries/carpentries-theme
+            if ! git merge -s recursive -Xtheirs --no-commit styles-ref
+            then
+
+              # Remove "deleted by us, unmerged" files from the staging area.
+              # these are the files that were removed from the lesson
+              # but are still present in the carpentries/styles repo
+              git rm $(git diff --name-only --diff-filter=DU)
+
+              # If there are still "unmerged" files,
+              # let's raise an error and look into this more closely
+              if [[ -n $(git diff --name-only --diff-filter=U) ]]
+              then
```

I think it would be beneficial to report the state of the repository that failed this test. 

```suggestion
              then
                echo "There were unmerged files in ${{ matrix.lesson-name }}:"
                echo "$(git diff --compact-summary --diff-filter=U)"
```